Mancala/src/MancalaBackend/MancalaBoard.java

```diff
@@ -42,6 +46,99 @@ public void startNewGame(PitSize size) {
     public PlayerTypes getTurn() {
         return turn;
     }
+    
+    /**
+     * Get the player A's pit list.
+     * @return player A pits
+     */
+    public MancalaPit[] getPlayerAPits() {
+    	return playerAPits;
+    }
+    
+    /**
+     * Get the player B's pit list.
+     * @return player B pits
+     */
+    public MancalaPit[] getPlayerBPits() {
+    	return playerBPits;
+    }
+    
+    /**
+     * Switch the turn.
+     */
+    public void nextTurn() {
+    	turn = turn == PlayerTypes.PlayerA ? PlayerTypes.PlayerB : PlayerTypes.PlayerA;
+    }
+    
+    /**
+     * The player who has turn pick up stones from the certain pit and placing them.
+     * @param targetPit - the index of pit that the player chose
+     * @return True if the player gains free turn, false if not.
+     */
+    public boolean pickUpStones(int targetPit) {
```

Comment:
before we start cowboy coding and adding the features to the mancala board, we should get together and plan out a structure that we agree on and follow.